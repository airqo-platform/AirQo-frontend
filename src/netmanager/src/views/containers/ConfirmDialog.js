import React from "react";
import { Button, Dialog } from "@material-ui/core";

import "assets/scss/confirm-dialog.sass";

const ConfirmDialog = ({
  open,
  close,
  confirmBtnMsg,
  confirm,
  title,
  message,
  error,
}) => {
  let style = { background: "#2f67e2", color: "white" };
  if (error) {
    style = { background: "#c00", color: "white" };
  }
  return (
    <Dialog
      open={open}
      onClose={close}
      aria-labelledby="confirm-dialog-title-del"
      aria-describedby="confirm-dialog-description"
      classes={"confirm-dialog"}
    >
      <div className={"confirm-dialog"}>
        <div className={"confirm-dialog-title"}>{title}</div>
        <div className={"confirm-dialog-message"}>{message}</div>
        <div className={"confirm-dialog-controls"}>
          <Button
            variant="contained"
            // style={{ background: "#c6d1db" }}
            onClick={close}
          >
            Cancel
          </Button>
          <Button
            variant="contained"
            type="submit"
            onClick={confirm}
            style={style}
          >
            {confirmBtnMsg || "Delete"}
          </Button>
        </div>
      </div>
    </Dialog>
  );
};

export default ConfirmDialog;
