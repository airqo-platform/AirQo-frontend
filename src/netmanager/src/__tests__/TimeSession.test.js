import React from 'react';
import { shallow } from 'enzyme';
import ConfirmDialog from 'views/containers/ConfirmDialog';

describe('ConfirmDialog Component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <ConfirmDialog
        open={true}
        close={() => {}}
        title="Session Expired"
        message="Your session has expired due to inactivity. Please log in again."
        confirmBtnMsg="Log In"
        confirm={() => {}}
        error={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
