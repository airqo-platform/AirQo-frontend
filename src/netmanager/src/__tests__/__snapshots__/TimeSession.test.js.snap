// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog Component renders without crashing 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-describedby="confirm-dialog-description"
  aria-labelledby="confirm-dialog-title-del"
  classes="confirm-dialog"
  onClose={[Function]}
  open={true}
>
  <div
    className="confirm-dialog"
  >
    <div
      className="confirm-dialog-title"
    >
      Session Expired
    </div>
    <div
      className="confirm-dialog-message"
    >
      Your session has expired due to inactivity. Please log in again.
    </div>
    <div
      className="confirm-dialog-controls"
    >
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
        variant="contained"
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
        style={
          Object {
            "background": "#2f67e2",
            "color": "white",
          }
        }
        type="submit"
        variant="contained"
      >
        Log In
      </WithStyles(ForwardRef(Button))>
    </div>
  </div>
</WithStyles(ForwardRef(Dialog))>
`;
